src/ArmTemplates/Extractor/EntityExtractors/PolicyFragmentsExtractor.cs

```diff
@@ -0,0 +1,75 @@
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Builders.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Policy;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.PolicyFragments;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.EntityExtractors.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.EntityExtractors
+{
+    public class PolicyFragmentsExtractor : IPolicyFragmentsExtractor
+    {
+        readonly ILogger<PolicyFragmentsExtractor> logger;
+        readonly IPolicyFragmentsClient policyFragmentsClient;
+        readonly ITemplateBuilder templateBuilder;
+
+        public PolicyFragmentsExtractor(
+            ILogger<PolicyFragmentsExtractor> logger,
+            ITemplateBuilder templateBuilder,
+            IPolicyFragmentsClient policyFragmentsClient)
+        {
+            this.logger = logger;
+            this.templateBuilder = templateBuilder;
+            this.policyFragmentsClient = policyFragmentsClient;
+        }
+
+        public async Task<Template<PolicyFragmentsResources>> GeneratePolicyFragmentsTemplateAsync(List<PolicyTemplateResource> apiTemplatePolicies, ExtractorParameters extractorParameters)
+        {
+            var policyFragmentsTemplate = this.templateBuilder
+                                        .GenerateTemplateWithApimServiceNameProperty()
+                                        .Build<PolicyFragmentsResources>();
+
+            var policyFragments = await this.policyFragmentsClient.GetAllAsync(extractorParameters);
+            if (policyFragments.IsNullOrEmpty())
+            {
+                this.logger.LogWarning($"No policy fragments were found for '{extractorParameters.SourceApimName}' at '{extractorParameters.ResourceGroup}'");
+                return policyFragmentsTemplate;
+            }
+
+            foreach (var policyFragment in policyFragments)
+            {
+                policyFragment.Type = ResourceTypeConstants.PolicyFragments;
+                policyFragment.Name = $"[concat(parameters('{ParameterNames.ApimServiceName}'), '/{policyFragment.Name}')]";
+                policyFragment.ApiVersion = GlobalConstants.ApiVersionPreview;
+
+                if (string.IsNullOrEmpty(extractorParameters.SingleApiName))
+                {
+                    policyFragmentsTemplate.TypedResources.PolicyFragments.Add(policyFragment);
+                }
+                else
+                {
+                    var policyFragmentReferenceString = $"fragment-id=\"{policyFragment.OriginalName}\"";
+                    var isReferencedInPolicy = apiTemplatePolicies?.Any(x => x.Properties.PolicyContent.Contains(policyFragmentReferenceString));
```

Comment:
There is existing api endpoint: https://docs.microsoft.com/en-us/rest/api/apimanagement/current-preview/policy-fragment/list-references?tabs=HTTP which allows us to fetch all resources where policyFragment is connected. 
However, in case when there will be many policy fragments, we should make additional API call for each of them to check if it is referenced or not. This is good from one source-of-truth point but adds additional extraction time (depends of course on the count of policy fragments) 
Having that the policy fragment when referenced should have $"fragment-id=\"{policyFragment}\ this exact string in the policy, apart from that to have more accurate results there is a possibility to parse the xml and verify that it is referenced using include-fragment xml element. 
Basically, there are two options: 
1 - add additional api call for each policy fragment -> which increases time of exporting the templates. 
2 - add additional xml element check 
Open to discussion

Reply:
I do agree with the source of truth to be one api call, but image scenario when one api is being extracted.
There 1000 policy fragments -> 1000 additional api calls having Api policy content is extracted and available for comparison without additional call. 
Regarding error prone: the search pattern was built using the documentation of policy-fragment resource, worst case scenario is when policy fragment will be extracted that is not used by API, in case there will be some custom xml which contains exact same value.